unstructured_inference/models/chipper.py

```diff
@@ -1,173 +1,502 @@
-SPECIAL_TAGS = ["<sep/>"] + [
-    f"<{open_or_close}s_{label_type}>" for label_type in LABEL_MAP for open_or_close in ("", "/")
-]
-
 
 class UnstructuredChipperModel(UnstructuredElementExtractionModel):
-    required_w: int = 1248
-    required_h: int = 1664
-
     def initialize(
         self,
-        tokenizer_name: str,
-        pre_trained_model_name: str,
+        pre_trained_model_repo: str,
+        swap_head: bool,
+        prompt: str,
+        max_length: int,
+        heatmap_h: int,
+        heatmap_w: int,
+        source: str,
+        no_repeat_ngram_size: int = 10,
         auth_token: Optional[str] = os.environ.get("UNSTRUCTURED_HF_TOKEN"),
     ):
         """Load the model for inference."""
-        self.tokenizer: AutoTokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
-
-        self.processor: DonutProcessor = DonutProcessor(
-            image_processor=DonutImageProcessor(
-                do_resize=True,
-                size=(self.required_w, self.required_h),
-            ),
-            tokenizer=self.tokenizer,
+        if torch.cuda.is_available():
+            self.device = "cuda"
+        else:
+            self.device = "cpu"
+        self.max_length = max_length
+        self.heatmap_h = heatmap_h
+        self.heatmap_w = heatmap_w
+        self.source = source
+        self.processor = DonutProcessor.from_pretrained(pre_trained_model_repo, token=auth_token)
+        self.tokenizer = self.processor.tokenizer
+        self.logits_processor = NoRepeatNGramLogitsProcessor(
+            no_repeat_ngram_size,
+            get_table_token_ids(self.processor),
         )
 
-        self.tokenizer.add_tokens(new_tokens=SPECIAL_TAGS, special_tokens=True)
-
         self.model = VisionEncoderDecoderModel.from_pretrained(
-            pre_trained_model_name,
+            pre_trained_model_repo,
             ignore_mismatched_sizes=True,
             use_auth_token=auth_token,
         )
+        if swap_head:
+            lm_head_file = hf_hub_download(
+                repo_id=pre_trained_model_repo,
+                filename="lm_head.pth",
+                token=auth_token,
+            )
+            rank = 128
+            self.model.decoder.lm_head = torch.nn.Sequential(
+                torch.nn.Linear(self.model.decoder.lm_head.weight.shape[1], rank, bias=False),
+                torch.nn.Linear(rank, rank, bias=False),
+                torch.nn.Linear(rank, self.model.decoder.lm_head.weight.shape[0], bias=True),
+            )
+            self.model.decoder.lm_head.load_state_dict(torch.load(lm_head_file))
+
+        self.input_ids = self.processor.tokenizer(
+            prompt,
+            add_special_tokens=False,
+            return_tensors="pt",
+        ).input_ids.to(self.device)
+
+        self.start_tokens = [
+            v
+            for k, v in self.processor.tokenizer.get_added_vocab().items()
+            if k.startswith("<s_") and v not in self.input_ids
```

Comment:
another constant?

Reply:
Made the prefix a parameter in [a591613](https://github.com/Unstructured-IO/unstructured-inference/pull/232/commits/a5916134cdecfdc3452a478592efbc6ebfc67b91).